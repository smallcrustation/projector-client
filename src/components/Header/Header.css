h1{
  text-align: center;
  font-size: 25px;
  margin: 0 10px ;
  padding: .4em 0;
  transition: .3s;
}

h1:hover{
  font-size: 30px;
  background: none;
}

header{
  border-bottom: 1px solid black;
  display: flex;
  justify-content: space-between;
  align-content: center;
}

.Header__h1:hover{
  background-color: white;
}
