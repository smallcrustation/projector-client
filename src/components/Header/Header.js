import React from 'react'
import {Link} from 'react-router-dom'

import Nav from '../Nav/Nav'

import './Header.css'

export default class Header extends React.Component{


  render(){
    return (
      <header className="Header">
        <Link to='/' className="Header__h1"><h1 className="Header__h1">Projector</h1></Link>
        <Nav />
      </header>
    )
  }
}