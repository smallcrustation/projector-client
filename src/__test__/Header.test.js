import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'

import Header from '../components/Header/Header'
import renderer from 'react-test-renderer'

describe('App Testing', () => {
  it('Renders without crashing', () => {
    const div = document.createElement('div')

    ReactDOM.render(<BrowserRouter><Header /></BrowserRouter>, div)

    ReactDOM.unmountComponentAtNode(div)
  })

  it('Renders with no crashing and matches snapshot', () => {
    const tree = renderer.create(<BrowserRouter><Header /></BrowserRouter>).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
