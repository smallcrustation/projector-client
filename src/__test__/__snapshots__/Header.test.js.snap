// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Testing Renders with no crashing and matches snapshot 1`] = `
<header
  className="Header"
>
  <a
    className="Header__h1"
    href="/"
    onClick={[Function]}
  >
    <h1
      className="Header__h1"
    >
      Projector
    </h1>
  </a>
  <nav
    className="logged-out-nav nav-bar"
  >
    <a
      className="link-btn"
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
    <a
      className="link-btn"
      href="/createAccount"
      onClick={[Function]}
    >
      Create
    </a>
  </nav>
</header>
`;
